hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/datanode/DataNode.java

```diff
@@ -1079,6 +1084,22 @@ private String reconfSlowIoWarningThresholdParameters(String property, String ne
     }
   }
 
+  private String reconfStripedReconstructionParameters(String property, String newVal)
+      throws ReconfigurationException {
+    String result = null;
+    try {
+      if (property.equals(DFS_DN_EC_RECONSTRUCTION_THREADS_KEY)) {
+        int size = (newVal == null ? DFS_DN_EC_RECONSTRUCTION_THREADS_DEFAULT :
+            Integer.parseInt(newVal));
+        result = Long.toString(size);
+        ecWorker.setStripedReconstructionPoolSize(size);
+      }
+      return result;
+    } catch (IllegalArgumentException e) {
```

Comment:
Catch NumberFormatException alse.

Reply:
@zeekling NumberFormatException inherits from IllegalArgumentException. So I think there are no need to catch NumberFormatException.